Overall module update

kaosat-dev opened the pull request:
This should address  a few basic issues : 

- [x]  switch codebase to javascript
- [x] update dependencies 
- [x] flesh out readme
- [x] ensure that module works correctly on recent versions of node.js (v4 LTS, v6 etc)

kaosat-dev committed:
chore(package): updated package dependencies etc

kaosat-dev committed:
feat(migration to js ): added new version of code , this time es6
- recreatring coffeescript code in js (no auto conversion)
- moving towards less object oriented code : class PwmDriver => makePwmDriver
- cleanup

kaosat-dev committed:
chore(license): update date in license

kaosat-dev committed:
docs(readme): first cleanup  & update pass to readme

kaosat-dev committed:
doc(README): minor tweak to readme

kaosat-dev committed:
chore(examples): added basic example (wip)

kaosat-dev committed:
feat(i2c wrapper): added a promise based wrapper around the i2c library

kaosat-dev committed:
feat(codebase update): updated the whole codebase:
- cleaned up
- uses I2c promise based wrapper
- fixed a few lingering issues (compared to original python lib)
- removed obsolete/uneeded code

EMCP commented on package.json:
```diff
@@ -21,9 +21,7 @@
-    "coffee-script": "1.3.3",
-    "i2c": "0.1.1",
+    "i2c": "0.2.3",
     "sleep": "~1.1.1"
```
this didn't work for me on latest raspbian and `node@6.9.x` without `sleep@~3.0.0` .. that resolved to `sleep@3.0.1`

EMCP reviewed (changes requested):
The examples are very appreciated.  Just wanted to point out I couldn't get your project to bundle until I added `sleep@3.0.0`

kaosat-dev committed:
chore(package): updated dependencies etc

kaosat-dev committed:
doc(README): minor readme tweaks

kaosat-dev committed:
docs(README): fleshed out Readme further:
- added version
- added table of contents
- added badges
- added install infos
- fleshed out api entries

kaosat-dev committed:
refactor(removed old codebase):

kaosat-dev committed:
chore(package): bumped package number

kaosat-dev committed:
doc(README): minor tweaks

kaosat-dev committed:
doc(README): more tweaks...

kaosat-dev committed:
chore(build): added es5 build:
- added babel dependencies
- added build script & co
- modified main package entry
- added dist /build folder
- updated example to point to right version of the lib

kaosat-dev commented:
@EMCP  Hi Erik ! 
Thanks for testing this 'on the fly' , I had not yet pushed the changes to the package.json file (now done), I am using the latest I2c & sleep packages don't worry :)
I will also add a transpiled version down to es5 (babel & co) so that older versions of node also remain supported.
Btw if this PR is running for you locally , could you please check if it is also stable:  for some reason the example 'servo sweep' works for me for 5 minutes then the servo becomes jittery. 
I currently lack a good 5v psu, so I sadly failed to test with more than one servo (both are jittery, won't move correctly)

kaosat-dev committed:
fix(debug flag): debug flag should be set to false by default

kaosat-dev committed:
fix(): removed a useless debug message
